lib/core/src/Cardano/Wallet.hs

```diff
@@ -297,187 +375,273 @@ createWallet (WalletLayer _ bp db _ _ _) wid wname s = do
             , delegation = NotDelegating
             }
     DB.createWallet db (PrimaryKey wid) checkpoint metadata $> wid
+  where
+    bp = ctx ^. typed @(BlockchainParameters t)
```

Comment:
I wonder is the type application even needed here, or can ghc work it out? :smiling_imp:

Reply:
It is needed to disambiguate the `t` at the very least. So in practice, this would be enough:

```hs
    bp = ctx ^. typed @(_ t)
```

But it reads quite poorly IMO :sweat_smile:

Reply:
I am also in favor of using explicit type annotation here